### PR TITLE
Pass props from withAuthenticator to Component

### LINKED ISSUE #837: Unable to use `withAuthenticator` HOC with Next.js
Unable to use `withAuthenticator` HOC with Next.js due to lack of props `Component` and `pageProps`

image

```json
"dependencies": {
    "@aws-amplify/ui-react": "^2.0.0",
    "aws-amplify": "^4.3.8",
    "next": "^12.0.4",
    "react": "^17.0.2",
    "react-dom": "^17.0.2"
  }
```

```js
// src/pages/_app.js
import { Amplify } from 'aws-amplify'
import { withAuthenticator } from '@aws-amplify/ui-react'
import '@aws-amplify/ui-react/styles.css'
import awsConfig from '../aws-exports'

Amplify.configure(awsConfig)

export default withAuthenticator(function MyApp({ Component, pageProps, ...rest }) {
  // `rest` contains { user, signOut }
  console.log({ Component }) // Component is undefined
  return <Component {...pageProps} />
})
```

As a note, the `Authenticator` component works as intended

```js
// src/pages/_app.js
import { Amplify } from 'aws-amplify'
import { Authenticator } from '@aws-amplify/ui-react'
import '@aws-amplify/ui-react/styles.css'
import awsConfig from '../aws-exports'

Amplify.configure(awsConfig)

export default function MyApp({ Component, pageProps }) {
  return (
    <Authenticator>
      {({ signOut, user }) => (
        <main>
          <h1>Hello {user.username}</h1>
          <button onClick={signOut}>Sign out</button>
          <Component {...pageProps} />
        </main>
      )}
    </Authenticator>
  )
}

```

### PR DESCRIPTION
**Pull request by ericclemmons**
*Issue #, if available:* Closes #837

*Description of changes:*

`withAuthenticator` didn't pass along props, but this PR passes it & validates it with a new `.feature` test.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .changeset/olive-glasses-bake.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+"next-example": patch
+"e2e": patch
```
Comment (wlee221) [inline .changeset/olive-glasses-bake.md:3]: These need to be removed 🙃

Comment (wlee221) [reply .changeset/olive-glasses-bake.md:3]:
```suggestion
```

Comment (wlee221) [reply .changeset/olive-glasses-bake.md:3]: im thinkg about a GitHub action checks your changeset doesn't have any private packages until https://github.com/changesets/bot/issues/44 is resolved.

File: .changeset/olive-glasses-bake.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+---
+"next-example": patch
```
Comment (ericclemmons) [inline .changeset/olive-glasses-bake.md]:
```suggestion
```

**Review by wlee221 — APPROVED** [review]

**Pull request merged**